cogeo_mosaic/backends/base.py

```diff
@@ -101,7 +101,7 @@ def assets_for_point(self, lng: float, lat: float) -> List[str]:
 
     @cached(
         TTLCache(maxsize=cache_config.maxsize, ttl=cache_config.ttl),
-        key=lambda self, x, y, z: hashkey(self.path, x, y, z),
+        key=lambda self, x, y, z: hashkey(self.path, x, y, z, self.mosaicid),
```

Comment:
unrelated to the SQLite backend, but while writing the tests I realized that we were caching this even when the mosaic changed... to fix it we can use the mosaicid (built from the metadata, e.g mosaic version) to make sure we call the function when the mosaic has been updated.